client/src/components/Companies/Facts.js

```diff
@@ -0,0 +1,179 @@
+}));
+
+const companyFacts = [
+    {
+        CompanyID: 0,
+        Heading: [],
+        Summary: [],
+        ID: []
+    }
+]
+
+export default function Facts(props) {
+    const [factCitation, setFactCitation] = useState([]);
+    const [showCitation, setShowCitation] = useState(false);
+    const [fact, setFact] = React.useState(companyFacts);
+    const [state, setState] = useState([])
+
+    useEffect(() => {
+        axios.get('/facts',
+            {
+                params: props.company
+            }
+        )
+            .then((resp) => {
+                let data = companyFacts;
+                let headingsarr = [];
+                let summaryarr = [];
+                let idarr = [];
+                resp.data.map(fact => {
+                    headingsarr.push(fact['Heading']);
+                    summaryarr.push(fact['Summary']);
+                    idarr.push(fact['ID']);
+                })
+                data[0]['Heading'] = headingsarr;
+                data[0]['Summary'] = summaryarr;
+                data[0]["ID"] = idarr;
+                setFact(data);
+                setState(resp.data);
+            })
+      },[props.company]);
+
+      const styles = style();
+
+    const FactCitations = (i) => {
+        if (factCitation.length != 0) {
+            return <div><i>{JSON.parse(factCitation)[0]["Title"][i]}</i>, {JSON.parse(factCitation)[0]["Author"][i]}{JSON.parse(factCitation)[0]["Author"][i] && <span>,</span>} {JSON.parse(factCitation)[0]["PublishingGroup"][i]}, {JSON.parse(factCitation)[0]["Date"][i]}{JSON.parse(factCitation)[0]["Pages"][i] && <span>,</span>} {JSON.parse(factCitation)[0]["Pages"][i]}</div>
+        }
+    }
+
+        const showCitations = async () => {
+            if (showCitation == false) setShowCitation(true);
+            if (showCitation == true) setShowCitation(false);
+            // console.log(showCitation);
+            // if (showCitation == false) setShowCitation(true);
+            // if (showCitation == true) setShowCitation(false);
+            // if (showCitation == -1) {
+            //     setShowCitation(j);
+            //     console.log(showCitation);
+            // }
+            // if (showCitation == j) {
+            //     setShowCitation(-1);
+            //     console.log(showCitation);
+            // }
+            let citationsarr = [];
+            let relidarr = [];
+            let authorarr = [];
+            let datearr = [];
+            let pubarr = [];
+            let titlearr = [];
+            let urlarr = [];
+            let pagesarr = [];
+            if (fact[0]['Heading'].length != 0) {
+                Promise.all(Object.entries(fact[0]['Heading']).map((heading, i) =>
+                    axios.get('/citationsFacts',
+                        {
+                            params: [props.company, fact[0]["ID"][i], 'F']
+                        }
+                    ).then(resp => {
+                        if (resp.data.length != 0) {
```

Comment:
With new backend APIs, we don't have relationalID anymore for citations.
You can define a new structure instead of using 'cite' from company.js.   
you can define this structure in a new file like utils.js for this structure should be shared with other components like scores and news.

Reply:
Also i love how you get 'cite' into your component using props.
Maybe just import from js file is better.